### ISSUE TITLE
Connection in same time, login in the same account

### ISSUE THREAD

**Issue by juliensl** [wekan/wekan#2534]
Server Setup Information:

    Did you test in newest Wekan?: yes
    Wekan version: 2.97
    Operating System: Linux
    Deployment Method(snap/docker/sandstorm/mongodb bundle/source): snap
    Http frontend if any (Caddy, Nginx, Apache, see config examples from Wekan GitHub wiki first): Caddy
    Node Version: 8.16.0

Hi!
I just find a new weird bug, it will be very funny to debug it ^^
I think that is linked with my precedent post #2525 .

Context: I asked to my team to log themselves on wekan using the OAuth connection.
They did it and the magic event happened, they was all on the same account ^^
When someone edited his full name, the full name was change for everybody.
The fun fact is, when I went on the admin people panel, nobody was here, no accounts were created.

I tried to reproduce the bug in localhost, this is the process:
  1) open one browser, connect myself on my website
  2) open a second browser, connect myself with another account
  3) go on wekan with both
  4) click on "open with oidc" with the first, click on with the second
  5) accept to log you for the first time with the first browser, accept for the second
  6) Boom! You are on the same account
  7) Try to change the Full Name for one, it will be changed for the other too
  8) open another brother and connect yourself with your admin account, you'll not see any other account
  9) change again the full name, it will be changed for the 3 accounts (including the admin one)

I activated the debug log with oidc, but nothing important appear, it seems to be a classical connection.

I asked them to connect themselves one after the other, slowly, and it works.

I don't know what can happen, it looks like if there is a time where wekan mix accounts, but it's very weird, we have different tokens, we should have different accounts.

Which variable wekan choose to dispatch the data ? Id or email ?

Thanks again everybody for this wonderful project :)

**Comment by xet7** [comment]
Hehe, that's funny :) keep on poking Wekan, these are great :)

With password login, Wekan uses username to login.
With OIDC or LDAP, Wekan uses email address to login.

I have been thinking when do we hit limits of Meteor web framework, so I will see can I fix these or do I change Wekan to use some other framework.

**Comment by xet7** [comment]
@eric-burel 

Cool, thanks! Yes I welcome all help, Wekan still has 500 issues open, growing faster than I have time to implement and fix just by myself.

For finding Wekan code, use find.sh script, it ignores generated directories:
```
cd wekan
./find.sh
```
Wekan login code is at:
- `wekan/packages/*` OIDC, LDAP etc
- `wekan/components/main/layouts.js` login form authentication
- `wekan/server/authentication.js` login code, some permissions
- `wekan/models/*` Wekan database schema, some permissions and REST API

Also:
- [Directory Structure](https://github.com/wekan/wekan/wiki/Directory-Structure)
- [Developer Docs](https://github.com/wekan/wekan/wiki/Developer-Documentation)
- [Developer Docs for Sandstorm](https://github.com/wekan/wekan-maintainer/wiki/Developing-Wekan-for-Sandstorm), some more developer info there

Currently Wekan development is at master branch with Meteor 1.8.1, so some of the developer info above could be outdated.

Debugging:
```
sudo snap set wekan debug=true
sudo snap logs wekan.wekan
sudo snap logs wekan.mongodb
sudo snap logs wekan.caddy

export DEBUG=true
docker logs wekan-app
docker logs wekan-db
```
And also on webbrowser inspect console

At Sandstorm web, sandstorm web ui has log button, it gives log popup.

Wekan dependencies are listed at:
- `wekan/.meteor/packages`
- `wekan/.meteor/versions`
- `wekan/package.json`

Node packages are downloaded when building to `wekan/node_packages` and maybe also some other directory that is listed at `wekan/rebuild-wekan.sh`.

Meteor is usually installed to `~/.meteor`, that is `/home/USERNAME/.meteor`. Meteor is about 2 GB Javascript full stack framework, kitchen sink included. It made possible for original Wekan author mquandalle to create Wekan with so little amount of code. Plus, me maintaining Wekan after him, since 2016-12, with incremental updates.

**Comment by xet7** [comment]
So if you add to debug code like `console.log('message')` then it most likely will be visible in webbrowser inspect console.

**Comment by eric-burel** [comment]
Hi, thanks for the detailed answer.
I've managed to setup a local env with oauth, however when running from the `master` branch the redirect_uri has to include `?close` at the end of the route, I don't know why. It does not happen with Snap install. This lead to issue with the code token, which is in `query.close?code` instead of `query.code` in the `getToken` function.
Do you have an idea what could cause this? It's not blocking though so that's okay

**Comment by xet7** [comment]
@eric-burel 

Some login code could be wrong, thanks a lot for looking into it !

**Comment by xet7** [comment]
@eric-burel 

Currently OIDC login works with popup login style, but it would be nice if also redirect style could be made to work, because I have been trying to implement to `wekan/client/components/main/layout.js` onrendered part to automatically trigger clicking OIDC button to autologin. Sometimes it works, sometimes it does not work, because of webbrowser prevents popups or iframe prevents clicks. I would like to trigger OIDC function without any Javascript click tricks and popups, so that it would work in all browser. Actual webpage is already logged in with OIDC so getting iframe what has Wekan OIDC button trigger automatically is what is missing. I'm also trying to trigger iframe that has RocketChat login button.

**Comment by xet7** [comment]
I am also getting some cross-frame errors when trying to get main webpage Javascript to trigger Wekan OIDC button. Nowadays browsers have so many protections so I don't have this  working yet.

**Comment by juliensl** [comment]
Hi guys, 

I didn't success to reproduce the bug again on my localhost server. 
Did you do some progress on this topic ? 

If we cannot reproduce it, maybe we can work on the bug #2525 , I think they are linked.

**Comment by xet7** [comment]
@juliensl 

Wekan old Snap version at devel branch uses old Meteor 1.6.x and MongoDB 3.2.x. 

Newest Wekan 3.x at master branch uses Meteor 1.8.1 and 4.x, and works with source and Docker.

There is huge difference with old Wekan Snap version and newest Wekan version, with used package version numbers, many new features, etc.

I have not yet had time to try to duplicate this issue, so I don't know does this bug exist.

**Comment by juliensl** [comment]
Thanks for your answer xet7.

It's nice! I didn't see that there is a new version.
I don't have any time to explore it now, but I'll test it soon.

I saw that it's not available for snap now, you're planning to fix it, are you ?

**Comment by xet7** [comment]
@juliensl 

Yes, I have been trying to fix Snap etc for whole this month #2533, making many tries and commits. Well, it's just about 4000 servers that have Wekan Snap installed worldwide https://snapcraft.io/wekan so of course it will be fixed and updated.

**Comment by juliensl** [comment]
Great!
Thank you xet7 for all your work :)

**Comment by xet7** [comment]
@juliensl 

I have problems making newest Wekan Snap build, so I did ask for help at https://github.com/wekan/wekan/issues/2533#issuecomment-517999731 . Meanwhile, if someone needs newest Wekan, it's possible to switch to Docker.

**Comment by juliensl** [comment]
Hi again !

So, I did a test, 4 people, 4 different accounts, first login for everybody and....... It works !
4 different accounts was created on Wekan ! No error in the logs !

So, It's not a perfect test, but I cannot reproduce the bug, the other linked bug is solved #2525 , the first time, I got the bug with only 2 people, and today it works for 4. So, I guess it's solved :smiley:  

Thank you for the update guys !
